tests/regressions/test_action_extract_slots_12314.py

```diff
@@ -0,0 +1,157 @@
+from rasa.shared.core.trackers import DialogueStateTracker
+from rasa.shared.core.domain import Domain
+from rasa.shared.core.events import UserUttered
+
+SENDER = "sender"
+
+BOT_DIRECTORY = (
+    Path(os.path.dirname(__file__)).parent.parent
+    / "data"
+    / "test_action_extract_slots_12314"
+)
+
+
+@pytest.fixture
+def model_file():
+    if not (BOT_DIRECTORY / "models" / "model.tar.gz").exists():
+        output_directory = TemporaryDirectory()
+        print(output_directory)
+
+        train(
+            domain=str(BOT_DIRECTORY / "domain.yml"),
+            config=str(BOT_DIRECTORY / "config.yml"),
+            training_files=str(BOT_DIRECTORY / "data"),
+            output=str(BOT_DIRECTORY / "models"),
+            fixed_model_name="model",
+        )
+
+    return str(BOT_DIRECTORY / "models" / "model.tar.gz")
+
+
+async def test_setting_slot_with_custom_action(model_file: Text):
+    """
+    Attemps the following conversation:
+    User: help me install a rsa token
+    Bot: What type of rsa token do you need?
+    User: hard
+    Bot: User has requested rsa token type of hard
+    User: help me install a hard rsa token
+    Bot: User has requested rsa token type of hard
+
+    The custom action should set the slot `rsa_token` to
+    the value `hard` in the second and third message.
+    A `SlotSet` event should be emitted in the third message.
+
+    From the Jira Comment by Chris,
+    https://rasahq.atlassian.net/browse/ATO-678?focusedCommentId=18017
+    """
+    agent = await load_agent(model_path=model_file)
+    output_channel = CollectingOutputChannel()
+    domain = Domain.from_path(BOT_DIRECTORY / "domain.yml")
+    action_server_url = "https://my-action-server:5055/webhook"
+    endpoint = EndpointConfig(action_server_url)
+    remote_action = action.RemoteAction("action_rsa_token", endpoint)
+
+    event = UserUttered("Hi")
+    tracker = DialogueStateTracker.from_events(
+        sender_id="test_id", evts=[event], slots=domain.slots
+    )
+
+    # Check that the bot asks for the type of RSA token
+    with aioresponses() as mocked:
+        mocked.post(
+            action_server_url,
+            payload={
+                "events": [
+                    {
+                        "event": "slot",
+                        "name": "rsa_token",
+                        "value": "unknown",
+                    }
+                ],
+                "responses": [],
+            },
+        )
+
+        # Send the first message
+        await agent.handle_message(
+            _build_user_message(output_channel, "help me install a rsa token")
+        )
+        await remote_action.run(
+            CollectingOutputChannel(),
+            TemplatedNaturalLanguageGenerator(domain.responses),
+            tracker,
+            domain,
+        )
+
+    assert output_channel.messages[-1] == {
+        "recipient_id": SENDER,
+        "text": "What type of rsa token do you need?",
+    }
```

Comment:
@ancalita Can you please help me figure out what's wrong here? The assertion fails with,
```
FAILED tests/regressions/test_action_extract_slots_12314.py::test_setting_slot_with_custom_action - AssertionError: assert {'recipient_i...type of None'} == {'recipient_i...do you need?'}
  Omitting 1 identical items, use -vv to show
  Differing items:
  {'text': 'User has requested rsa token type of None'} != {'text': 'What type of rsa token do you need?'}
  Full diff:
  - {'recipient_id': 'sender', 'text': 'What type of rsa token do you need?'}
  ?                                     ^   ---- ^^^           ^ ^^ --  ---
  + {'recipient_id': 'sender', 'text': 'User has requested rsa token type of None'}
  ?                                     ^^^^^  ++++++++  ^           ^^^^^ ^^^
```

The bot instead uttered `'User has requested rsa token type of None'` this happens when the custom action specified in domain.yaml is not run.

Reply:
I think what's needed is to specify to the agent its `action_endpoint` [attribute](https://github.com/RasaHQ/rasa/blob/6f2a055d467a7139ecf65ebcc15d17acd66fbb2a/rasa/core/agent.py#L239):
```
agent.action_endpoint = endpoint
```

Then the agent can run the `RemoteAction` without you having to execute it in the test code. Hope that works 🤞🏻

Reply:
@ancalita I tried that it still fails, do you have any other ideas?

Reply:
Ah, I'm out of ideas, then maybe we can't test this without having a proper integration test with an instance of an action server running. I'd say this might require more effort, so if the manual tests validated the solution, let's not spend more time on the integration test, sorry for setting you on wild goose chase 😞

Reply:
@ancalita I think this was a great idea but I'm not sure at this point how to make this test work. Can I mark it as flaky and keep it there with some TODO comments?

Reply:
I think the flaky workflows have required status on `main`, so you'd have to comment out the failing assertions to prevent the test from failing. I also think that flaky marker doesn't describe this test well, because it always fails in current form 😄 Therefore I'd not add the flaky marker.

Reply:
@ancalita In that case, I've removed the regression tests from this PR. Can you please review it again?

Reply:
Sure, could you also please add a ticket in Jira maintenance epic for writing up an integration test with an instance of the action server, linking to this PR? 🙏🏻

Reply:
Yes, I've added a ticket here: https://rasahq.atlassian.net/browse/ATO-1046